File: ClassProblems/liftProblem.java
public class LiftController {
    private boolean[][] liftCalled;

    public LiftController() {
        liftCalled = new boolean[5][2];
        for (int i = 0; i < 5; i++) {
            liftCalled[i][0] = false;
            liftCalled[i][1] = false;
        }
    }

    public void supervisorCall(int floor) {
        for (int iFloor = 0; iFloor < liftCalled.length; iFloor++) {
            for (int iDirection = 0; iDirection < liftCalled[iFloor].length; iDirection++) {
                if (iFloor == floor) {
                    liftCalled[iFloor][iDirection] = true;
                } else {
                    liftCalled[iFloor][iDirection] = false;
                }
            }
        }
    }
}

File: ClassProblems/liftProblem.py
class LiftController:
    def __init__(self):
        self.lift_called = [[False,False] for _ in range(5)]
    def supervisor_call(self,floor):
        for i_floor in range(len(self.lift_called)):
            for i_direction in range(len(self.lift_called[i_floor])):
                if i_floor == floor:
                    self.lift_called[i_floor][i_direction] = True
                else:
                    self.lift_called[i_floor][i_direction] = False
    